docs/current/sdk/go/959738-get-started.md

```diff
@@ -75,356 +80,440 @@ func build(repoUrl string) error {
-
-## Step 3: Copy the build output to the host machine
-
-Once the tool has completed the build, it should put that build artifact somewhere to be used. In this step, Dagger will copy the build artifact to the host machine after the build is complete.
-
-1. Update the file `main.go` and some new steps to the `build()` function as shown below. Save the file once done.
-
-```go
-func build(repoUrl string) error {
-  ...
-  // 1. reference to the current working directory on the host
-  workdir := client.Host().Workdir() // <-- New
 
+  // get `golang` image
   golang := client.Container().From("golang:latest")
+
+  // mount cloned repository into `golang` image
   golang = golang.WithMountedDirectory("/src", src).WithWorkdir("/src")
 
-  // 2. Create the output path on the host for the build
-  // -->
+  // build application
   path := "build/"
-  outpath := filepath.Join(".", path)
-  err = os.MkdirAll(outpath, os.ModePerm)
-  if err != nil {
-    return err
-  }
-  // <-- New
-
   golang = golang.Exec(dagger.ContainerExecOpts{
     Args: []string{"go", "build", "-o", path},
   })
 
-  // 3. Get build output from builder
-  // -->
+  // get reference to build output directory in container
   output, err := golang.Directory(path).ID(ctx)
   if err != nil {
     return err
   }
-  // <-- New
 
-  // 4. Write the build output to the host
-  // -->
+  // create build/ directory on host
+  outpath := filepath.Join(".", path)
+  err = os.MkdirAll(outpath, os.ModePerm)
+  if err != nil {
+    return err
+  }
+
+  // get reference to current working directory on the host
+  workdir := client.Host().Workdir()
+
+  // write contents of container build/ directory
+  // to the host working directory
   _, err = workdir.Write(ctx, output, dagger.HostDirectoryWriteOpts{Path: path})
+  // highlight-end
   if err != nil {
     return err
   }
-  // <-- New
 
   return nil
 }
 ```
 
-With this new code, the tool will now write the build artifact to the host after the build is complete.
+The revised `build()` function is the main workhorse here, so let's step through it in detail.
 
-- using the Dagger Go SDK, a reference to the host workdir is created with `.Host().Workdir()`
-- in native Go, create a directory where the build artifact will be output
-- create a reference to the build output in the Dagger engine with `Container.Directory().ID()`
-- write the directory to the host with `HostDirectory.Write()`
+- It begins by creating a Dagger client with `dagger.Connect()`, as before.
+- It uses the client's `Git()` function to obtain a reference to the target repository. This function returns a `GitRepository` struct. The struct's `Branch()` function provides details on a specific branch (here, the `main` branch), the `Tree()` function returns the filesystem of the branch, and the `ID()` function returns a reference for the filesystem. This reference is stored in the `src` variable.
```

Comment:
```suggestion
- It uses the client's `Git()` function to obtain a reference to the target repository. This function returns a `GitRepository` struct. The struct's `Branch()` function provides details on a specific branch (here, the `main` branch), the `Tree()` function returns the directory of the branch, and the `ID()` function returns a reference for the directory. This reference is stored in the `src` variable.
```

Technically we are using the term Directory instead of filesystem in our API now (our old API used filesystem), so I think it's probably more consistent to use that terminology here and in the rest of this section too? Debatable though, as it's not like "filesystem" is really inaccurate, just slightly inconsistent maybe.